### DISCUSSION TITLE
New to Gitleaks - need help on extending allowList.

### DISCUSSION THREAD

**Discussion by thomaslicrc** [zricethezav/gitleaks#1558] [Q&A]
Hello,

I'm new to Gitleaks and haven't used regex in a long while.

I need to allow for this kind of values:
```
KeyToken=1234567891234567
KeyToken=1234567891234567"
KeyToken="1234567891234567"
```

I generated this regex:
```
[allowlist]
  regexes = [
    '''(?i)[^\w]*(KeyToken)=["']?[a-fA-F0-9]{16}["']?''', 
  ]
```

And my complete file looks like this (.gitleaks.toml):
```
[extend]
useDefault = true

[allowlist]
  regexes = [
    '''(?i)[^\w]*(KeyToken)=["']?[a-fA-F0-9]{16}["']?''', 
  ]
  
```
 But I still get this kind of results:
```
 "Description": "Generic API Key",
  "StartLine": 5794,
  "EndLine": 5794,
  "StartColumn": 131,
  "EndColumn": 156,
  "Match": "KeyToken=1234567891234567\"",
```
  

I launch gitleaks this way:
`gitleaks detect --source=repo_temp --config=$config_file_path --report-format=json --report-path=$report_file 
`

Am I missing something?

Thank you!

**Comment by thomaslicrc** [comment]
I also tried with a way simpler regex:

`    '''(?i)[^\w]*(KeyToken)=.*''',  # Allow anything after KeyToken=
`

**Comment by rgmz** [comment]
> But I still get this kind of results:

What is the value of `"secret"`? I suspect your issue is that Gitleaks' allowlist checks the _secret_ by default. From the README:

> ```
> # note: (global) regexTarget defaults to check the _Secret_ in the finding.
> # if regexTarget is not specified then _Secret_ will be used.
> # Acceptable values for regexTarget are "match" and "line"
> regexTarget = "match"
> ```

If that's the case it explains why your regex isn't working: `(?i)[^\w]*(KeyToken)=.*` does not match `1234567891234567`.

**Comment by thomaslicrc** [comment]
Oh OK!

The value of the secret is alphanumerical, 16 length.

I'll try to do the regex on the secret.
